### ISSUE TITLE
Support chunked execution

### ISSUE THREAD

**Issue by brendt** [spatie/async#34]
@sebastiandedeyne noticed a huge performance improvement when there are a lot of small tasks ran in parallel, by chunking them together. This way there can be a balance between the process creation overhead and the large amount of small tasks to be done.

This should be a configurable method on the `Pool`:

```php
Pool::create()
    ->tasksPerProcess(20);
```

**Comment by brendt** [comment]
Because this is such an edge case, I'm going to close this issue for now.

**Comment by zelin** [comment]
For some reason i am unable to create a new issue. Kindly have a look at this error i am receiving while adding entries to a pool

```
#0 Pushbot/src/Pushbot/Async/Process/ParallelProcess.php(126): Pushbot\Async\Output\ParallelError::fromException('PHP Warning:  M...')
#1 Pushbot/src/Pushbot/Async/Process/ProcessCallbacks.php(51): Pushbot\Async\Process\ParallelProcess->resolveErrorOutput()
#2 Pushbot/src/Pushbot/Async/Pool.php(201): Pushbot\Async\Process\ParallelProcess->triggerError()
#3 Pushbot/src/Pushbot/Async/Pool.php(297): Pushbot\Async\Pool->markAsFailed(Object(Pushbot\Async\Process\ParallelProcess))
#4 Pushbot/src/Pushbot/Async/Pool.php(128): Pushbot\Async\Pool->Pushbot\Async\{closure}(20, Array)
#5 Pushbot/src/Pushbot/Pushbot.php(421): Pushbot\Async\Pool->wait()
```

**Comment by brendt** [comment]
@zelin please see https://github.com/spatie/async/issues/36
